src/Optimizer.ts

```diff
@@ -0,0 +1,85 @@
+import { Report } from './Models/Report';
+import { AsyncAPIDocument, parse } from '@asyncapi/parser';
+import { RemoveComponents } from './Optimizers/RemoveComponents';
+import { MoveToComponents } from './Optimizers/MoveToComponents';
+import { ReuseComponents } from './Optimizers/ReuseComponents';
+import { Options } from './Models/Options';
+import { ReportElement } from './Models/Report';
+import YAML from 'yaml';
+const merge = require('merge-deep');
+const _ = require('lodash');
```

Comment:
You should be consistent and use the ESM:

```suggestion
import merge from 'merge-deep';
import * as _ from 'lodash';
```